packages/rulesets/src/oas/functions/oasExample.ts

```diff
@@ -110,6 +110,54 @@ function* getSchemaValidationItems(
   }
 }
 
+/**
+ * Modifies 'schema' (and all its sub-schemas) to remove all "example" fields.
+ * In this context, "sub-schemas" refers to all schemas reachable from 'schema'
+ * (e.g. properties, additionalProperties, allOf/anyOf/oneOf, not, items, etc.)
+ * @param schema the schema to be "de-examplified"
+ * @returns 'schema' with example fields removed
+ */
+function deExamplify(schema: Record<string, unknown>): Record<string, unknown> {
```

Comment:
have you consider using `traverse` from `json-schema-traverse`?
This would help simplify the traversal.

I'm happy to look into that since we kept you waiting for quite a while already (apologies for that!)

Reply:
Hi @P0lip , I was not aware of json-schema-traverse when I worked on the PR so I didn't consider using that.   Please feel free to make follow-on changes to improve the function.